app/js/features/escrow/saga.js

```diff
@@ -132,8 +197,20 @@ export function *rateTx({escrowId, rating}) {
   try {
     const toSend = Escrow.methods.rateTransaction(escrowId, rating);
     const estimatedGas = yield call(toSend.estimateGas);
-    yield call(toSend.send, {gasLimit: estimatedGas + 1000, from: web3.eth.defaultAccount});
-    yield put({type: RATE_TRANSACTION_SUCCEEDED, escrowId, rating});
+    const promiseEvent = toSend.send({gasLimit: estimatedGas + 1000, from: web3.eth.defaultAccount});
+    const channel = eventChannel(promiseEventEmitter.bind(null, promiseEvent));
+    while (true) {
```

Comment:
Do you think we could we wrap that into a function?
It seems we repeat always the same pattern when we send a transaction

Reply:
Sadly no, because the `yields` need to be in the same scope as the original function. It's really the big problem with the current way sagas work, you can't call actions from within another function and we need them because it uses events.

Reply:
ohhh :cry: I see what you mean, this is also quite annoying.
Do you think we could make them generic enough to have like:
https://github.com/embark-framework/embark/blob/master/embark-ui/src/sagas/index.js#L11-L19

For all transaction call?